test/uk/gov/hmrc/pillar2stubs/controllers/BTNControllerCrossPeriodSpec.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2stubs.controllers
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Helpers}
+import uk.gov.hmrc.pillar2stubs.models.btn.BTNRequest
+
+import java.time.LocalDate
+
+class BTNControllerCrossPeriodSpec extends AnyWordSpec with Matchers {
+
+  private val controller = new BTNController(Helpers.stubControllerComponents(), null, null)
+
+  "BTNController" when {
+
+    "handling XEPLR9999999995 BTN submissions" should {
+
+      "apply Scenario 3 logic for Period 2 (currentYear - 2) submission" in {
```

Comment:
Could you please make these test names a little bit more descriptive? I see references to Scenarios 3 & 4 in the ticket, but the scenarios themselves have now been removed. This might be the right behaviour, but I have no way of checking the stubbed data matches what the test wants like this.